### ISSUE TITLE
Hackers Stole Access Tokens from Okta’s Support Unit

### ISSUE THREAD

**Issue by swidup** [vz-risk/VCDB#20018]
https://krebsonsecurity.com/2023/10/hackers-stole-access-tokens-from-oktas-support-unit/

https://thehackernews.com/2023/10/oktas-support-system-breach-exposes.html

https://www.cnbc.com/2023/10/20/okta-shares-fall-after-company-says-client-files-were-accessed-by-hackers-via-its-support-system.html

**Comment by swidup** [comment]
https://www.infosecurity-magazine.com/news/okta-reveals-breach-via-stolen/

**Comment by swidup** [comment]
https://blog.1password.com/okta-incident/ 1Password affected

**Comment by swidup** [comment]
https://sec.okta.com/harfiles

"Unauthorized Access to Okta's Support Case Management System: Root Cause and Remediation

Executive Summary

We offer our apologies to those affected customers, and more broadly to all our customers that trust Okta as their identity provider. We are deeply committed to providing up-to-date information to all our customers.

On Thursday, October 19, Okta advised customers of a security incident. Having finalized our investigation, we can confirm that from September 28, 2023 to October 17, 2023, a threat actor gained unauthorized access to files inside Okta’s customer support system associated with 134 Okta customers, or less than 1% of Okta customers. Some of these files were HAR files that contained session tokens which could in turn be used for session hijacking attacks. The threat actor was able to use these session tokens to hijack the legitimate Okta sessions of 5 customers, 3 of whom have shared their own response to this event.

The unauthorized access to Okta’s customer support system leveraged a service account stored in the system itself. This service account was granted permissions to view and update customer support cases. During our investigation into suspicious use of this account, Okta Security identified that an employee had signed-in to their personal Google profile on the Chrome browser of their Okta-managed laptop. The username and password of the service account had been saved into the employee’s personal Google account. The most likely avenue for exposure of this credential is the compromise of the employee’s personal Google account or personal device. 

Failure to identify file downloads in customer support vendor logs

For a period of 14 days, while actively investigating, Okta did not identify suspicious downloads in our logs. When a user opens and views files attached to a support case, a specific log event type and ID is generated tied to that file. If a user instead navigates directly to the Files tab in the customer support system, as the threat actor did in this attack, they will instead generate an entirely different log event with a different record ID.

Okta’s initial investigations focused on access to support cases, and subsequently we assessed the logs linked to those cases. On October 13, 2023, BeyondTrust provided Okta Security a suspicious IP address attributed to the threat actor. With this indicator, we identified the additional file access events associated with the compromised account.

Investigation Timeline

2023-09-29 1Password reports suspicious activity to Okta Support.

2023-09-29 Okta Security begins an investigation, suspecting that 1Password was most likely the victim of malware or a phishing attack. 

2023-09-29 to 2023-10-02 Okta Security meets with 1Password on 9/29, 9/30, 10/1 and 10/2 in an attempt to resolve their support case.  

2023-10-02 BeyondTrust reports suspicious activity to Okta Support.

2023-10-02 to 2023-10-11 Okta Security meets with 1Password and BeyondTrust multiple times from 10/2 to 10/11. 

2023-10-12 A third customer reports suspicious activity to Okta Support.

2023-10-13 BeyondTrust provides Okta Security an indicator of compromise (IP address) associated with the event they reported to Okta Support on 10/2.

2023-10-16 Using the supplied IP address, Okta Security identifies a service account associated with previously unobserved events in the customer support system logs.

2023-10-17 Okta Security disables the service account and terminates associated sessions.

2023-10-17 Okta Security copies and examines all files identified in the customer support system logs that were accessed by the threat actor. 134 Okta customers or less than 1% of Okta customers had a file accessed by the threat actor. 

2023-10-17 Okta Security revokes the Okta session tokens embedded in the HAR files.

2023-10-17 Okta Security investigates whether the threat actor attempted to access customer Okta instances using these files.

2023-10-18 Okta Security notifies a fourth Okta customer targeted by the adversary.

2023-10-18 Okta Security identified a gap in the logs from the customer support system, missing the final hours that the threat actor had access. A re-run query now returns a complete picture of adversary activity. 

2023-10-19 Okta Security identifies additional files downloaded by the threat actor that were not previously discovered due to the delay in receiving the logs.

2023-10-19 Okta Security revokes the Okta session tokens embedded in the newly discovered HAR files that had been downloaded by the threat actor.

2023-10-19 Okta Security identifies Cloudflare as the fifth and final Okta target of the adversary.

2023-10-19 Okta alerts all Okta customers with registered security contacts, confirming if they were or were not impacted by the security incident.  

2023-10-20 Okta publishes public advisory at https://sec.okta.com/articles/2023/10/tracking-unauthorized-access-oktas-support-system 

2023-10-20 to 2023-11-02 Okta is focused on helping all customers, answering their questions and rolling out remediation steps.

2023-11-02 Okta notifies all Okta customers with registered security contacts of the root cause and remediation steps.

2023-11-03 Okta publishes root cause and remediation steps at https://sec.okta.com/harfiles.  

Remediation Tasks

1. Disabled the compromised service account (Complete)
Okta has disabled the service account in the customer support system. 

2. Blocking the use of personal Google profiles with Google Chrome (Complete)
Okta has implemented a specific configuration option within Chrome Enterprise that prevents sign-in to Chrome on their Okta-managed laptop using a personal Google profile.

3. Enhanced monitoring for the customer support system (Complete)

Okta has deployed additional detection and monitoring rules for the customer support system.

4. Binding Okta administrator session tokens based on network location (Complete)

Okta has released session token binding based on network location as a product enhancement to combat the threat of session token theft against Okta administrators. Okta administrators are now forced to re-authenticate if we detect a network change. This feature can be enabled by customers in the early access section of the Okta admin portal."

https://techcrunch.com/2023/10/24/oktas-latest-hack-fallout-hits-cloudflare-1password/
